File: python/5 kyu/390.py
# https://www.codewars.com/kata/525c65e51bf619685c000059/train/python

# Passed

def cakes(recipe, available):
    result = 0
    can_bake_cakes = True
    while can_bake_cakes:
        for ingredient in recipe:
            quantity_required = recipe[ingredient]
            available[ingredient] = available.get(ingredient, 0) - quantity_required
            if available[ingredient] < 0:
                can_bake_cakes = False
                break
        else:
            result += 1
            
    return result

output = cakes({"flour": 500, "sugar": 200, "eggs": 1}, {"flour": 1200, "sugar": 1200, "eggs": 5, "milk": 200})
print(output)

File: nodejs/7 kyu/390.js
// https://www.codewars.com/kata/563cf89eb4747c5fb100001b/train/javascript

// Passed

function removeSmallest(numbers) {
  let minIdx = 0;
  for (let idx = 0; idx < numbers.length; idx++) {
    if (numbers[idx] < numbers[minIdx]) {
      minIdx = idx;
    }
  }

  let result = numbers.slice(0, minIdx).concat(numbers.slice(minIdx + 1));
  return result;
}

const output = removeSmallest([2, 2, 1, 2, 1]);
console.log(output);